### PR TITLE
fix(api) Fix tests broken by PR #599

### PR DESCRIPTION
**Pull request by rjuliano**
*Issue #, if available:*

*Description of changes:*
My changes for #599 ended up causing some subscription related tests to fail. This PR fixes that.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-api/src/main/java/com/amplifyframework/api/aws/SubscriptionEndpoint.java [hunk 1]
```diff
@@ -103,6 +102,14 @@
         Objects.requireNonNull(onSubscriptionError);
         Objects.requireNonNull(onSubscriptionComplete);
 
+        if (subscriptionUrl == null) {
+            try {
+                subscriptionUrl = buildConnectionRequestUrl();
+            } catch (ApiException apiException) {
+                onSubscriptionError.accept(apiException);
```
Comment (jamesonwilliams) [inline aws-api/src/main/java/com/amplifyframework/api/aws/SubscriptionEndpoint.java:109]: Should you add `return;` after this line, to cede the flow of control from `requestSubscription(...)`?

Comment (rjuliano) [reply aws-api/src/main/java/com/amplifyframework/api/aws/SubscriptionEndpoint.java:109]: Yikes...good eye on that one. I debated just making the function throw, but it started betting messy upstream.

File: aws-api/src/main/java/com/amplifyframework/api/aws/SubscriptionEndpoint.java [hunk 2]
```diff
@@ -107,10 +105,16 @@
...
-            webSocket = okHttpClient.newWebSocket(new Request.Builder()
-                .url(subscriptionUrl)
-                .addHeader("Sec-WebSocket-Protocol", "graphql-ws")
-                .build(), webSocketListener);
+            try {
+                webSocket = okHttpClient.newWebSocket(new Request.Builder()
+                    .url(buildConnectionRequestUrl())
+                    .addHeader("Sec-WebSocket-Protocol", "graphql-ws")
+                    .build(), webSocketListener);
+            } catch (ApiException apiException) {
```
Comment (jamesonwilliams) [inline aws-api/src/main/java/com/amplifyframework/api/aws/SubscriptionEndpoint.java:113]: What actually throws the `ApiException`? The `buildConnectionRequestUrl()` bit? The rest looks likes its pure OkHttp, and wouldn't throw one of our internal types.

Comment (rjuliano) [reply aws-api/src/main/java/com/amplifyframework/api/aws/SubscriptionEndpoint.java:113]: Yeah...it's buried deep down into the code that adds the security headers.

**Pull request merged**